torch/csrc/autograd/functions/special.cpp

```diff
@@ -264,11 +265,10 @@ bool Eval::replaceSubgraph(const variable_list& inputs, const variable_list& _ou
     // This output is already rebased. This happens when there
     // the same Variable has been returned multiple times, and
     // is repeated in this list.
-    if (output.get()->_grad_fn.get() == this) {
+    if (output.grad_fn().get() == this) {
```

Comment:
This change may have performance implications: `grad_fn()` will update the _grad_fn if output is a view with a stale _grad_fn. I don't think that's what you want for this comparison.

Reply:
Thanks, I've created a `grad_fn_ptr()` class which returns the raw pointer.

Reply:
How about `grad_fn_unsafe()`? It's unclear what's the difference between `grad_fn` and `grad_fn_ptr`, while it really is unsafe to use the second method without knowing what you're doing